scripts/upsert/sections.ts

```diff
@@ -0,0 +1,53 @@
+import { Term } from '../../lib/term';
+import { differenceInMinutes } from 'date-fns';
+import { upsertSections } from '../../lib/sections';
+import { findLatestTask, createTask } from '../../lib/task';
+import { Fetch } from '../../lib/banner/fetch';
+import makeFetchCookie from 'fetch-cookie';
+import { getSearchResults } from '../../lib/banner';
+import { range } from '../../lib/fn';
+import fetch from 'node-fetch';
+
+export const upsertSectionsScript = async (term: string, registrationDay: Date, dropDate: Date) => {
+  const today = new Date();
+
+  const lastUpdated = await findLatestTask(`upsertSections-${term}`);
+
+  const minutesSinceLastUpdate = differenceInMinutes(today, lastUpdated?.startedAt ?? -1);
+
+  // If after drop date and courses have been updated within the last 12 hours, return
+  if (today > dropDate && minutesSinceLastUpdate <= 720) return;
+
+  // Else, run the task
+
+  // Establish a session for each request
+  const fc = makeFetchCookie(fetch) as unknown as Fetch;
+
+  // Get sections
+  const init = await getSearchResults(fc, { term, max: 500 });
+  let sectionsCount = 0;
+  const { totalCount } = init;
+
+  sectionsCount += init.data.length;
+  const offsets = range(init.data.length, totalCount, init.pageMaxSize);
+
+  const data = [...init.data];
+  for (const offset of offsets) {
+    const { data } = await getSearchResults(fc, { term, max: init.pageMaxSize, offset });
+    data.concat(data);
+    sectionsCount += data.length;
+  }
+
+  await createTask(`upsertSections-${term}`, upsertSections(fc, term, data), {});
+};
+
+if (process.env[2] && process.env[3]) {
+  const registrationDate = new Date(process.argv[2]);
```

Comment:
We probably want some default values?

Reply:
Yeah I was thinking that and I had some at first, but then changed it so that the scripts will _only_ run if the arguments are provided because I wasn't sure which dates to pick (I assume most current term?). Let me know what you think and I can add defaults